app/controllers/events_controller.rb

```diff
@@ -11,17 +9,19 @@ def all
   end
 
   def create
-    event = Apple.new.fire({ message: params[:message] })
-    publish event
 
-    event = Orange.new.receive event
-    publish event
+    originating_data = { message: params[:message] }
+
+    steps = [{ name: 'Apple',  type: 'EventHandler' },
+             { name: 'Orange', type: 'EventHandler' },
+             { name: 'Banana', type: 'EventHandler' },
+             { name: 'Pear',   type: 'EventHandler' }]
```

Comment:
@calebcauthon It's more than simpler.  Do you see how we've gone from concrete code to data?

Now the next step is to make an editor that builds JSON like this.  Another screen that uses vis.js, but someone can build a workflow that takes pre-existing event handlers (we just have one, "EventHandler") and allows them to string them together however they wish.